### PR TITLE
client, tests: testify the client tests

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: ref #4813.

### What is changed and how does it work?

```commit-message
Testify the client tests.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/client_test.go [hunk 1]
```diff
@@ -16,38 +16,27 @@ package pd
...
+func TestTsLessEqual(t *testing.T) {
+	re := require.New(t)
+	re.True(tsLessEqual(9, 9, 9, 9))
+	re.True(tsLessEqual(8, 9, 9, 8))
+	re.False(tsLessEqual(9, 8, 8, 9))
+	re.False(tsLessEqual(9, 8, 9, 6))
+	re.True(tsLessEqual(9, 6, 9, 8))
 }
 
-func TestMain(m *testing.M) {
```
Comment (rleungx) [inline client/client_test.go:34]: We don't need it anymore?

Comment (JmPotato) [reply client/client_test.go:34]: Ops, it seems I deleted it accidentally.

File: tests/client/client_test.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright 2018 TiKV Project Authors.
+// Copyright 2018 TiKV Project Authortc.
```
Comment (rleungx) [inline tests/client/client_test.go:1]:
```suggestion
// Copyright 2018 TiKV Project Authors.
```

File: tests/client/client_test.go [hunk 3]
```diff
@@ -575,7 +592,7 @@ func (i *idAllocator) alloc() uint64 {
 var (
 	regionIDAllocator = &idAllocator{allocator: &mockid.IDAllocator{}}
 	// Note: IDs below are entirely arbitrary. They are only for checking
-	// whether GetRegion/GetStore works.
+	// whether GetRegion/GetStore worktc.
```
Comment (rleungx) [inline tests/client/client_test.go:595]:
```suggestion
	// whether GetRegion/GetStore works.
```

File: tests/client/client_test.go [hunk 4]
```diff
@@ -617,38 +634,41 @@ type testClientSuite struct {
...
+	})
 	checker.IsNil = func(obtained interface{}) {
-		c.Assert(obtained, IsNil)
+		re.Nil(obtained)
 	}
 	return checker
 }
 
-func (s *testClientSuite) SetUpSuite(c *C) {
+func newTestClient(re *require.Assertions) *testClient {
```
Comment (rleungx) [inline tests/client/client_test.go:647]: We can also use suite here.

Comment (JmPotato) [reply tests/client/client_test.go:647]: The tests inside a suite can not be run in parallel, see https://github.com/stretchr/testify/issues/187.

Comment (JmPotato) [reply tests/client/client_test.go:647]: After consideration, let's keep using suite here to finish the migration first.

**Review by bufferflies — APPROVED** [review]

File: tests/client/client_test.go [hunk 5]
```diff
@@ -938,125 +1010,125 @@ func (s *testClientSuite) TestGetRegionByID(c *C) {
...
 	// Get an offline store should be OK.
-	n, err = s.client.GetStore(context.Background(), store.GetId())
-	c.Assert(err, IsNil)
-	c.Assert(n, DeepEquals, offlineStore)
+	n, err = suite.client.GetStore(context.Background(), store.GetId())
+	suite.NoError(err)
+	suite.True(reflect.DeepEqual(offlineStore, n))
 
-	// Should return offline stores.
+	// Should return offline storetc.
```
Comment (rleungx) [inline tests/client/client_test.go:1058]:
```suggestion
	// Should return offline stores.
```

File: tests/client/client_test.go [hunk 6]
```diff
@@ -938,125 +1010,125 @@ func (s *testClientSuite) TestGetRegionByID(c *C) {
...
+	n, err = suite.client.GetStore(context.Background(), physicallyDestroyedStoreID)
+	suite.NoError(err)
 	if n != nil { // store is still offline and physically destroyed
-		c.Assert(n.GetNodeState(), Equals, metapb.NodeState_Removing)
-		c.Assert(n.PhysicallyDestroyed, IsTrue)
+		suite.Equal(metapb.NodeState_Removing, n.GetNodeState())
+		suite.True(n.PhysicallyDestroyed)
 	}
-	// Should return tombstone stores.
+	// Should return tombstone storetc.
```
Comment (rleungx) [inline tests/client/client_test.go:1083]: ditto

**Review by rleungx — APPROVED** [review]

**Pull request merged**